__tests__/app.test.js

```diff
@@ -12,6 +12,90 @@ afterAll(() => {
+      return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({body: {endpoints}}) => {
+        expect(typeof endpoints).toBe("object")
+        expect(Object.keys(endpoints).length).toBeGreaterThan(0)
+      })
+    })
+    it("every every endpoint to have a METHOD and an endpoint", () => {
+      return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({body: {endpoints}}) => {
+        const arrayOfEndpoints = Object.keys(endpoints)
+        arrayOfEndpoints.forEach((endpoint) => {
+          expect(typeof endpoint).toBe('string')
+          expect(/[A-Z]*/.test(endpoint)).toBe(true)
+          expect(/\/.*[\w]$/.test(endpoint)).toBe(true)
+        })
+      })
+    })
+    it("expects every endpoint to have a description", () => {
+      return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({body: {endpoints}}) => {
+        const arrayOfEndpoints = Object.keys(endpoints)
+        arrayOfEndpoints.forEach((endpoint) => {
+          const {description} = endpoints[endpoint]
+          expect(typeof description).toBe('string')
+        })
+      })
+    })
+    it("expects every endpoint to have an array of allowed queries", () => {
+      return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({body: {endpoints}}) => {
+        const arrayOfEndpoints = Object.keys(endpoints)
+        arrayOfEndpoints.forEach((endpoint) => {
+          const {queries} = endpoints[endpoint]
+          expect(Array.isArray(queries)).toBe(true)
+          queries.forEach((query) => {
+            expect(typeof query).toBe('string')
+          })
+        })
+      })
+    })
+    it("expects every endpoint to have a string format if given one", () => {
+      return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({body: {endpoints}}) => {
+        const arrayOfEndpoints = Object.keys(endpoints)
+        arrayOfEndpoints.forEach((endpoint) => {
+          const {format} = endpoints[endpoint]
+          if (format) expect(typeof format).toBe('string')
+        })
+      })
+    })
+    it("expects every endpoint to have an example response object that contains responses", () => {
+      return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({body: {endpoints}}) => {
+        const arrayOfEndpoints = Object.keys(endpoints)
+        arrayOfEndpoints.forEach((endpoint) => {
+          const {exampleResponse} = endpoints[endpoint]
+          expect(typeof exampleResponse).toBe('object')
+          const exampleResponseKeys = Object.keys(exampleResponse)
+          expect(exampleResponseKeys.length).toBeGreaterThan(0)
+          exampleResponseKeys.forEach((key) => {
+            const responseContainsObject = exampleResponse[key].length > 0 || Object.keys(exampleResponse[key]).length > 0
+            expect(responseContainsObject).toBe(true)
+          })
+        })
+      })
+    })
+  })
+})
```

Comment:
Sorry Nathan. All you need to do here is require in the endpoints json and compare it to what is returned by your server. Should only need one test here.